pep-0690.rst

```diff
@@ -209,38 +403,73 @@ This PEP proposes to add a new ``importlib.eager_imports()`` context manager,
 so the above technique can be less verbose and doesn't require comments to
 clarify its intent::
 
+    from importlib import eager_imports
+
     with eager_imports():
         import foo
         import bar
 
 Since imports within context managers are always eager, the ``eager_imports()``
 context manager can just be an alias to a null context manager. The context
-manager does not force all imports to be recursively eager: ``foo`` and ``bar``
-will be imported eagerly, but imports within those modules will still follow
-the usual laziness rules.
+manager's effect is not transitive: ``foo`` and ``bar`` will be imported
+eagerly, but imports within those modules will still follow the usual laziness
+rules.
 
 The more difficult case can occur if an import in third-party code that can't
-easily be modified must be forced to be eager. For this purpose, we propose to
-add an API to ``importlib`` that can be called early in the process to specify
-a list of module names within which all imports will be eager::
+easily be modified must be forced to be eager. For this purpose,
+``importlib.enable_lazy_imports()`` takes an optional keyword-only ``eager``
+argument, which can be set to a container of module names within which all
+imports will be eager::
 
-    from importlib import set_eager_imports
+    from importlib import enable_lazy_imports
 
-    set_eager_imports(["one.mod", "another"])
+    enable_lazy_imports(eager=["one.mod", "another"])
```

Comment:
I have some questions about this change.

Later on, the PEP describes what the semantics are for calling this function twice: the second call _replaces_ the names from the first call.  To me then `set` is a better name than `enable` because the latter might imply that it augments the original set of names, while the former is clearly replacing any original names.

This change also says that the `eager` keyword is optional, but it doesn't describe what the syntax or semantics of the function is if you don't give the argument.  Or am I missing that description in the PEP?  The implication to me is that if not given, then _all_ imports are lazy from this point forward.  IOW, it's a way to mimic the effects of the `-L` option at runtime.  The PEP should be explicit about that (if it isn't already).

I suspect that the argument version will be more common than the no-argument version, which could lead to even more confusion, because it has both "lazy" and "eager" in the same call site.  I think a better keyword argument name would be `excluding` (we obviously can't use `except` here).

Thus, to programmatically enable all imports be lazy:

```py
set_lazy_imports()
```

and to set all imports to be lazy except a few:

```py
set_lazy_imports(excluding=['one.mod', 'another'])
# or
set_lazy_imports(excluding=eager_import_matcher)
```

Does this deserve more discussion on the [Discourse thread](https://discuss.python.org/t/pep-690-lazy-imports/15474/39)?

Reply:
I'm happy with the name change to `set_lazy_imports` and `excluding` kwarg, as well as the clarification about what happens with no arg; will add all that.

> Does this deserve more discussion on the [Discourse thread](https://discuss.python.org/t/pep-690-lazy-imports/15474/39)?

Sure, but at this point I think the most productive way to prompt that conversation is to propose something in the PEP and then toss the new PEP draft out for a new round of feedback, not to ask this specific question before updating the PEP.

Reply:
I think the behavior of `set_lazy_imports()` with no argument was already clarified at the very beginning of the Specification section:

```
Lazy imports are opt-in, and globally enabled either via a new ``-L`` flag to
the Python interpreter or via a call to a new ``importlib.set_lazy_imports()``
function, which makes all imports after the call lazy.
```

But I also added clarification of this in the opt-out section where the `excluding` arg is described.